components/fxa-client/src/profile.rs

```diff
@@ -27,6 +27,7 @@ impl FirefoxAccount {
                         "Access token rejected, clearing the tokens cache and trying again."
                     );
                     self.clear_access_token_cache();
+                    self.clear_devices_and_attached_clients_cache();
```

Comment:
This clear here might not be as useful as the other two, but might still be worth keeping it here.